crates/modelardb_server/src/metadata/compressed_file.rs

```diff
@@ -0,0 +1,322 @@
+    /// Size of the file in bytes.
+    pub(crate) size: usize,
+    /// Timestamp that the file was created at.
+    pub(super) created_at: Timestamp,
+    /// Timestamp of the first data point in the file.
+    pub(super) start_time: Timestamp,
+    /// Timestamp of the last data point in the file.
+    pub(super) end_time: Timestamp,
+    /// Value of the data point with the smallest value in the file.
+    pub(super) min_value: Value,
+    /// Value of the data point with the largest value in the file.
+    pub(super) max_value: Value,
+}
+
+impl CompressedFile {
+    pub fn new(
+        name: Uuid,
+        folder_path: PathBuf,
+        size: usize,
+        created_at: Timestamp,
+        start_time: Timestamp,
+        end_time: Timestamp,
+        min_value: Value,
+        max_value: Value,
+    ) -> Self {
+        Self {
+            name,
+            folder_path,
+            size,
+            created_at,
+            start_time,
+            end_time,
+            min_value,
+            max_value,
+        }
+    }
+
+    /// Convert the given file information and [`RecordBatch`] to a [`CompressedFile`].
+    pub(crate) fn from_record_batch(
+        name: Uuid,
+        folder_path: PathBuf,
+        size: usize,
+        batch: RecordBatch,
+    ) -> Self {
+        // unwrap() is safe since UNIX_EPOCH is always earlier than now.
+        let since_the_epoch = SystemTime::now().duration_since(UNIX_EPOCH).unwrap();
+        let created_at = since_the_epoch.as_millis() as Timestamp;
+
+        // unwrap() is safe as None is only returned if all of the values are None.
+        let start_time =
+            aggregate::min(modelardb_common::array!(batch, 2, TimestampArray)).unwrap();
+        let end_time = aggregate::max(modelardb_common::array!(batch, 3, TimestampArray)).unwrap();
+
+        // unwrap() is safe as None is only returned if all of the values are None.
+        // Both aggregate::min() and aggregate::max() consider NaN to be greater than other non-null
+        // values. So since min_values and max_values cannot contain null, min_value will be NaN if all
+        // values in min_values are NaN while max_value will be NaN if any value in max_values is NaN.
+        let min_value = aggregate::min(modelardb_common::array!(batch, 5, ValueArray)).unwrap();
+        let max_value = aggregate::max(modelardb_common::array!(batch, 6, ValueArray)).unwrap();
+
+        Self {
+            name,
+            folder_path,
+            size,
+            created_at,
+            start_time,
+            end_time,
+            min_value,
+            max_value,
+        }
+    }
+
+    /// Extract the file name from each [`ObjectMeta`] in `object_metas` and convert it to an UUID,
+    /// if possible. If any [`ObjectMeta`] is not an Apache Parquet file with an UUID file name,
+    /// return [`String`].
+    pub(crate) fn object_metas_to_compressed_file_names(
+        object_metas: Vec<ObjectMeta>,
```

Comment:
Maybe replace `Vec` with a slice `&[OjbectMeta]` to not force the caller to copy the `object_metas` and then make another copy of the data in this method when returning `Vec<Uuid>`.

Reply:
Fixed.